superdesk/types/users.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+"""Superdesk Users"""
+
+
+from datetime import datetime
+from typing import Annotated, Any
+
+from pydantic import Field
+from superdesk.core.resources import ResourceModel, fields
+from superdesk.core.resources.fields import ObjectId
+from superdesk.core.resources.validators import validate_unique_value_async, validate_data_relation_async
+
+
+class UsersResourceModel(ResourceModel):
+    username: Annotated[fields.Keyword, validate_unique_value_async("users", "username")]
+    password: str = Field(min_length=5)
+    password_changed_on: datetime | None = None
+    first_name: str
+    last_name: str
+    display_name: str
+    email: Annotated[fields.Keyword, validate_unique_value_async("users", "email")]
+    phone: str | None = None
+    job_title: str | None = None
+    biography: str | None = None
+    facebook: str | None = None
+    instagram: str | None = None
+    twitter: str | None = None
+    jid: Annotated[fields.Keyword, validate_unique_value_async("users", "jid")] | None = None
+    language: str | None = None
+    user_info: dict[str, Any] = Field(default_factory=dict)
+    picture_url: str | None = None
+    avatar: Annotated[ObjectId, validate_data_relation_async("uploads")] | None = None
+    avatar_renditions: dict[str, Any] | None = Field(default=None)
+    role: Annotated[ObjectId, validate_data_relation_async("roles")]
+    privileges: dict[str, Any] = Field(default_factory=dict)
+    workspace: dict[str, Any] = Field(default_factory=dict)
+    user_type: str = Field(default="user")
```

Comment:
This should be an enum, with possible values `["user", "administrator", "external"]`